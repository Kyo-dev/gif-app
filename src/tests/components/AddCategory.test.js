import "@testing-library/jest-dom";
import React from "react";
const { shallow } = require("enzyme");
const { AddCategory } = require("../../components/AddCategory");

describe("test in <AddCategory />", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  // snapshot
  test("Show component OK", () => {
    expect(wrapper).toMatchSnapshot();
  });
  // input event
  test("should change input text", () => {
    const input = wrapper.find("input");
    const value = "hola mundo";
    input.simulate("change", {
      target: {
        value,
      },
    });
    expect(wrapper.find("p").at(0).text().trim()).toBe(value);
  });

  test("should't post data ", () => {
    wrapper.find("form").simulate("submit", {
      preventDefault() {},
    });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Should call setCategories and clean inputText", () => {
    const value = "Hola";
    wrapper.find("input").simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
