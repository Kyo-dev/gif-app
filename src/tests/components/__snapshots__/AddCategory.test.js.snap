// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <AddCategory /> Show component OK 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</form>
`;
